Sources/MapboxNavigation/OrnamentsController.swift

```diff
@@ -73,42 +73,29 @@ extension NavigationMapView {
                                                    name: UIDevice.orientationDidChangeNotification,
                                                    object: nil)
             NotificationCenter.default.addObserver(self,
-                                                   selector: #selector(didUpdateEHorizonRoadName),
-                                                   name: .electronicHorizonDidUpdatePosition,
+                                                   selector: #selector(didUpdateRoadNameFromStatus),
+                                                   name: .routeControllerRoadName,
                                                    object: nil)
-
         }
         
         private func suspendNotifications() {
             NotificationCenter.default.removeObserver(self,
                                                       name: UIDevice.orientationDidChangeNotification,
                                                       object: nil)
             NotificationCenter.default.removeObserver(self,
-                                                      name: .electronicHorizonDidUpdatePosition,
+                                                      name: .routeControllerRoadName,
                                                       object: nil)
         }
         
-        @objc func didUpdateEHorizonRoadName(_ notification: Notification) {
-            guard let startingEdge = notification.userInfo?[RoadGraph.NotificationUserInfoKey.treeKey] as? RoadGraph.Edge else { return }
-            let edgeIdentifier = startingEdge.identifier
-            
-            guard let routeController = navigationViewData.router as? RouteController,
-                  let metadata = routeController.roadGraph.edgeMetadata(edgeIdentifier: edgeIdentifier) else { return }
-            
-            electronicHorizonRoadNames = metadata.names.map { name -> String in
-                switch name {
-                case .name(let name):
-                    return name
-                case .code(let code):
-                    return code
-                }
-            }
-        }
-        
         @objc func orientationDidChange(_ notification: Notification) {
             updateMapViewOrnaments()
         }
         
+        @objc func didUpdateRoadNameFromStatus(_ notification: Notification) {
+            guard let roadName = notification.userInfo?[RouteController.NotificationUserInfoKey.roadNameKey] as? String else { return }
```

Comment:
We should always update `roadNameFromStatus` even if it’s null. Otherwise, if the user travels from a named road to an unnamed road, the previous road’s name could incorrectly persist in the label.

Reply:
Thanks for the suggestion, the tail work is addressed in #3266 .